components/brave_new_tab_ui/containers/newTab/settings/braveToday/useManageDirectFeeds.ts

```diff
@@ -56,43 +60,109 @@ export default function useManageDirectFeeds (publishers?: Publishers) {
   }
   const [feedInputText, setFeedInputText] = React.useState<string>()
   const [feedInputIsValid, setFeedInputIsValid] = React.useState<FeedInputValidity>(FeedInputValidity.Valid)
+  const [feedSearchResults, setFeedSearchResults] = React.useState<FeedSearchResultModel[]>([])
   const onChangeFeedInput = (e: React.ChangeEvent<HTMLInputElement>) => {
     setFeedInputText(e.target.value)
     setFeedInputIsValid(FeedInputValidity.Valid)
   }
-  const onAddSource = React.useCallback(async () => {
-    // Validate
+
+  const onSearchForSources = React.useCallback(async () => {
```

Comment:
why are you using a `useCallback` here?

Reply:
Addressed via https://github.com/brave/brave-core/compare/4957cf379b..f3cb1e9232